src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -64,136 +55,99 @@ public CreateCaseCallbackService(
+        if (eventValidation.isInvalid()) {
+            return new ProcessResult(emptyList(), singletonList(eventValidation.getError()));
+        } else {
+            return hasServiceNameInCaseTypeId(request.getCaseDetails())
+                .map(serviceName -> {
+                    try {
+                        ServiceConfigItem serviceCfg = serviceConfigProvider.getConfig(serviceName);
+                        if (serviceCfg == null || serviceCfg.getTransformationUrl() == null) {
+                            return new ProcessResult(
+                                emptyList(),
+                                singletonList("Transformation URL is not configured")
+                            );
+                        } else {
+                            return validator
+                                .getValidation(request.getCaseDetails())
+                                .map(exceptionRecord -> {
+                                    String exceptionRecordId = request.getCaseDetails().getId().toString();
+
+                                    SuccessfulTransformationResponse transformationResp =
+                                        transformToCase(
+                                            exceptionRecord,
+                                            serviceCfg,
+                                            exceptionRecordId
+                                        );
+
+                                    if (!transformationResp.warnings.isEmpty() && !request.isIgnoreWarnings()) {
+                                        return new ProcessResult(transformationResp.warnings, emptyList());
+                                    } else {
+                                        long newCaseId = createNewCaseInCcd(
+                                            idamToken,
+                                            userId,
+                                            exceptionRecord.poBoxJurisdiction,
+                                            transformationResp.caseCreationDetails,
+                                            exceptionRecordId
+                                        );
+                                        return new ProcessResult(ImmutableMap.of(
+                                            CASE_REFERENCE,
+                                            Long.toString(newCaseId)
+                                        ));
+                                    }
+                                })
+                                .getOrElseGet(errors -> new ProcessResult(emptyList(), errors.asJava()));
+                        }
+                    } catch (ServiceNotConfiguredException exc) {
+                        return new ProcessResult(emptyList(), singletonList(exc.getMessage()));
+                    } catch (InvalidCaseDataException exc) {
+                        return new ProcessResult(exc.getResponse().warnings, exc.getResponse().errors);
```

Comment:
I can see the old code was checking if the response was a 400 or not - after the change it doesn't. Only after 422 we should return a response like this. In all other cases it should be an exception thrown, so that CCD would return 500.